spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/job/support/StepExecutionProgressInfo.java

```diff
@@ -64,7 +64,7 @@ public StepExecutionProgressInfo(StepExecution stepExecution, StepExecutionHisto
 		if (startTime == null) {
 			startTime = LocalDateTime.now();
 		}
-		duration = Duration.between(startTime, endTime).get(ChronoUnit.MILLIS);
+		duration = Duration.between(startTime, endTime).get(ChronoUnit.NANOS);
 		percentageComplete = calculatePercentageComplete();
```

Comment:
I notice the `calculatePercentageComplete` is called after duration is set. It is a bit weird that the only call to `getDurationBasedEstimate` accepts the duration as parameter and in other uses references the field.